### DISCUSSION TITLE
Performance of loading master css

### DISCUSSION THREAD

**Discussion by ghost** [master-co/css#86] [Question]
1) Would the script 'https://cdn.master.co/css' affect the page's loading time?
2) I have downloaded the script and checked its size, it's "49kb". But the doc said it's  "~14KB" ( https://docs.master.co/css/setup ), do I misunderstand the doc or am I missing something?

**Comment by 1aron** [comment]
[![Package Size](https://flat.badgen.net/badgesize/brotli/https://cdn.jsdelivr.net/npm/@master/css?icon=jsdelivr&label&color=yellow)](https://unpkg.com/@master/css 'brotli package size (without dependencies)')

1. External CDN will not be the first choice, if you can import `@master/css` into your local project and deliver your site through CDN service, it'll be the best practice.

2. The 49KB size of JIT is uncompressed size.
 
[Brotli](https://en.wikipedia.org/wiki/Brotli) is primarily used by [web servers](https://en.wikipedia.org/wiki/Web_servers) and [content delivery networks](https://en.wikipedia.org/wiki/Content_delivery_networks) to [compress HTTP content](https://en.wikipedia.org/wiki/HTTP_compression), making internet websites load faster. A successor to [gzip](https://en.wikipedia.org/wiki/Gzip), it is supported by all major web browsers and has become increasingly popular, as it provides better compression than gzip.

In the upcoming v2.0, we propose a new compilation architecture, which allows you to compile the CSS required by the current page ahead of time, and load the JIT engine package asynchronously ( It'll not be included in the first load ).

**Comment by ghost** [comment]
Hi, "JIT engine package of Master CSS" means the script? Do you mean I download it and put it into my hosting then use Brotli to compress it then use my CDN to deliver it?

**Reply by 1aron** [reply]
## Setup JIT in your project
Install to your project:
```bash
npm install @master/css@1
```
Import to your entry file like `main.js`:
```bash
import '@master/css'
```

Resources are usually compressed over the browser network, and you don't need to do anything. You can observe some resource size changes by inspecting DevTools -> Network.
